core/src/main/java/nl/nn/adapterframework/receivers/Receiver.java

```diff
@@ -1256,33 +1256,37 @@ private Message processMessageInAdapter(Object rawMessageOrWrapper, Message mess
 					pipeLineResult = adapter.processMessageWithExceptions(businessCorrelationId, pipelineMessage, pipelineSession);
 					setExitState(threadContext, pipeLineResult.getState(), pipeLineResult.getExitCode());
 					pipelineSession.put("exitcode", ""+ pipeLineResult.getExitCode());
-					result=pipeLineResult.getResult().asString();
-					if(result != null && result.length() > ITransactionalStorage.MAXCOMMENTLEN) {
-						errorMessage = "exitState ["+pipeLineResult.getState()+"], result ["+result.substring(0, ITransactionalStorage.MAXCOMMENTLEN)+"]";
-					}else {
-						errorMessage = "exitState ["+pipeLineResult.getState()+"], result ["+result+"]";
+					result=pipeLineResult.getResult();
+
+					errorMessage = "exitState ["+pipeLineResult.getState()+"], result [";
+					if(!Message.isEmpty(result) && result.size() > ITransactionalStorage.MAXCOMMENTLEN) { //Since we can determine the size, assume the message is preservable
+						errorMessage += result.asString().substring(0, ITransactionalStorage.MAXCOMMENTLEN);
```

Comment:
Hier wil je niet dat Message.size() zo smart is om de lengte van een file te weten, en dan hier eerst een hele file van zeg 1GB in geheugen te lezen om er maar een klein stukje van af te drukken.
Er moet dus naar mijn idee ook nog een test bij op result.asObject() instanceof String